lib/W3/Request.php

```diff
@@ -121,14 +130,19 @@ static function get_as_array($prefix, $default = array()) {
      * @return array
      */
     static function get_request() {
-        if (!isset($_GET)) {
-            $_GET = array();
-        }
-
-        if (!isset($_POST)) {
-            $_POST = array();
-        }
 
-        return array_merge($_GET, $_POST);
+    	if( !self::$request ){
```

Comment:
Correct me if im wrong but i think this line might need to be changed to:
```php 
if( !isset(self::$request) ){
```
The reason being that if both the _$GET and $_POST are empty arrays (ie, array() ) then the array_merge into self::request will also be an empty array and so the `if( !self::$request ){` will always return false, allowing it to enter causing it to set itself every time.